jquery.flot.js

```diff
@@ -1670,40 +1693,94 @@ Licensed under the MIT license.
+                        if (axis.position == "top") {
+                            yoff = -yoff;
+                        }
+                    } else {
+                        y = axis.p2c(v);
+                        xoff = t;
 
-                if (axis.direction == "x") {
-                    x = axis.p2c(v);
-                    yoff = t;
+                        if (axis.position == "left") {
+                            xoff = -xoff;
+                        }
+                    }
 
-                    if (axis.position == "top")
-                        yoff = -yoff;
-                } else {
-                    y = axis.p2c(v);
-                    xoff = t;
+                    if (ctx.lineWidth == 1) {
+                        if (axis.direction == "x")
+                            x = Math.floor(x) + 0.5;
+                        else
+                            y = Math.floor(y) + 0.5;
+                    }
 
-                    if (axis.position == "left")
-                        xoff = -xoff;
-                }
+                    ctx.moveTo(x, y);
+                    ctx.lineTo(x + xoff, y + yoff);
+                }
+
+                //draw minor tick marks
+                if(minorTicks == true && i < axis.ticks.length - 1)
+                {
+                    var minorTicksNr = 4; // default number of minor ticks
+                    var v1 = axis.ticks[i].v,
+                        v2 = axis.ticks[i + 1].v,
+                        step = (v2 - v1) / (minorTicksNr + 1),
+                        xminor = 0,
+                        yminor = 0,
+                        j;
+
+                    for(j = 1; j <= minorTicksNr; j++)
+                    {
+                        // compute minor tick position
+                        if(axis.direction == "x")
+                        {
+                            yminor = t / 2; // minor ticks are half length
+                            x = axis.p2c(v1 + j*step);
+
+                            if (ctx.lineWidth == 1) {
+                                x = Math.floor(x) + 0.5;
+                            }
 
-                if (ctx.lineWidth == 1) {
-                    if (axis.direction == "x")
-                        x = Math.floor(x) + 0.5;
-                    else
-                        y = Math.floor(y) + 0.5;
-                }
+                            if (axis.position == "top") {
+                                yminor = -yminor;
+                            }
 
-                ctx.moveTo(x, y);
-                ctx.lineTo(x + xoff, y + yoff);
+                            // don't go over the plot borders
+                            if((x < 0) || (x > plotWidth))
+                                continue;
+                        } else {
+                            xminor = t / 2; // minor ticks are half length
+                            y = axis.p2c(v1 + j*step);
+
+                            if (ctx.lineWidth == 1) {
```

Comment:
Could you please define a local function variable here that does this correction and reused it for both x and y? Something like:
var crispy = function(d) {
  return ctx.lineWidth === 1 ? Math.floor(d) + 0.5 : d;
}